### PR TITLE
Fix for translation of OpFMod instruction

### PR DESCRIPTION
**Pull request by kkyzylova**
OpFMod(a, b) should be translated to copysign(frem(a, b), b)

### CODE REVIEW

**Comment by bader** [comment]
LGTM.

**Comment by yxsamliu** [comment]
I am not sure about this.

This change translates SPIRV OpFMod(x,y) to copysign(frem(x,y),y) where frem is llvm frem instruction, and copysign is opencl builtin.

Two questions:
1. llvm frem returns the remainder of a division. The remainder has the same sign as the dividend. Why do we still need copysign?
2. frem(-2.1, -2) should get -0.1 since that's the remainder with the same sign.
   SPIRV FMod is defined as floating-point modulo operation of Operand 1 modulo Operand 2. The sign of a non-0 result comes from Operand 2.

My understanding is that modulo of -2.1 to -2 is 1.9. By forcing the sign of dividend, we should get -1.9.

Am I missing something? Or should we ask the WG for clarification about OpFMod? Thanks.

**Comment by kkyzylova** [comment]
1. In this case the remainder should have the same sign as the divisor. So
2. According to Raun Krisch:

>    OpFRem(x,y) == OpenCL.ext.fmod(x,y)
>    OpFMod(x,y) == copysign( OpenCL.ext.fmod(x,y), y )

frem(-2.1, -2) should get -0.1, yes. And the result of OpFMod is -0.1 too: copysign(-0.1, -2) = -0.1.

**Comment by yxsamliu** [comment]
I see. So
OpFRem <=> llvm.frem
OpenCL.ext.fmod <=> OpenCL.ext.fmod (equivalent to llvm.frem)
OpFMod => OpenCL.ext.copysign(llvm.frem)

File: test/SPIRV/fmod.ll [hunk 1]
```diff
@@ -1,51 +0,0 @@
-; __kernel void fmod_kernel( float out, float in1, float in2 )
```
Comment (yxsamliu) [inline test/SPIRV/fmod.ll:1]: I think we need to keep this test but modify it to be bi-way test and match expected result. I think it should be translated to extended instruction OpenCL.fmod, right?

Comment (kkyzylova) [reply test/SPIRV/fmod.ll:1]: Done

**Pull request merged**